src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -265,6 +265,11 @@ export function TopNavMenu(props: TopNavMenuProps): ReactElement | null {
                     </EuiFlexItem>
                     <EuiFlexItem className="globalDatePicker">
                       <div ref={datePickerRef} />
+                      {!showDatePicker && props.customSubmitButton && (
```

Comment:
why wouldn't it be similar to         showDatePicker={![TopNavMenuItemRenderType.OMITTED, false].includes(showDatePicker!)}


and it's a little confusing that the custom submit button would  be dependendent on a date picker being present or not.

if it is shouldn't it be apart of the the date picker?